### ISSUE TITLE
[Feature Request] Enable/Disable Logs For Specific Clients Or Client Groups

### ISSUE THREAD

**Issue by Cybo1927** [AdguardTeam/AdGuardHome#1326]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [X] I am running the latest version
- [ ] I checked the documentation and found no answer
- [ ] I checked to make sure that this issue has not already been filed

### Problem Description
I use AdGuard Home through my router, I have it setup with parental controls for my sister and tracking/ad blocking for me. But, my dad doesn't want his stuff being logged or filtered and I don't want to see it because it's not important to me. I only check logs for seeing if my sisters stuff is being blocked correctly or blocking ads in Windows apps.

### Proposed Solution
Add a button in the Client Settings to enable/disable logs for the selected client

**Comment by ArchiveBase** [comment]
Unfortunately, I dont like about disable logs for specific clients feature, instead we can consider about middle ground solution with additional enhancement https://github.com/AdguardTeam/AdGuardHome/issues/916 (remove client IP addresses from logs) while enable/disable logs feature already exist.

**Comment by ameshkov** [comment]
Well, it's either #916 or this, I guess we should choose.

**Comment by Cybo1927** [comment]
My request was mostly to do with parental controls and my dad just wanting to see what my sister is doing (he doesn't approve of things like Snapchat and refuses to let her get it) and he wants to make sure it's blocked and she didn't find a way around it. He's always not allowed most social media because he doesn't want his stuff being tracked because of one of us. I had told him this also works as a network wide ad/tracking blocker and so now we use it in the router but he doesn't want to see his own stuff or even mine, just my little sister's.

I don't know if that changes anything but another request could be to make it so we can search by client groups instead of having to use a bunch of different IP addresses.

**Comment by ArchiveBase** [comment]
Just a comment....

> But, my dad doesn't want his stuff being logged or filtered and I don't want to see it because it's not important to me.
> but he doesn't want to see his own stuff or even mine, just my little sister's.
> he doesn't want his stuff being tracked because of one of us.
> and he wants to make sure it's blocked and she didn't find a way around it.
> My request was mostly to do with parental controls and my dad just wanting to see what my sister is doing (he doesn't approve of things like Snapchat and refuses to let her get it)
> I only check logs for seeing if my sisters stuff is being blocked correctly or blocking ads in Windows apps.

I think below features may fulfill those most things you described. It seems hand over super administrator access to your father might be also better while you get limited access for necessary devices under custom permissions. (however it doesn't prevent you to hold an administrator account or manage any other devices)

- https://github.com/AdguardTeam/AdGuardHome/issues/997 User Accounts 
- https://github.com/AdguardTeam/AdGuardHome/issues/1245 Roles & Permissions (doubt if that implement)

Enable/disable logs for the selected client does not guarantee that you can't re-enable it, I believe he dont want to trust you 😄 

> another request could be to make it so we can search by client groups instead of having to use a bunch of different IP addresses.

https://github.com/AdguardTeam/AdGuardHome/issues/1273

**Comment by Cybo1927** [comment]
@ArchiveBase I actually do think that stuff is better than what I was proposing. Thanks!

@ameshkov With what ArchiveBase has shown I believe it's best to go with #916 instead of what I was proposing. Issues #997 #1245 and #1273 would do all that I needed. I believe that it's more important for people using AdGuard Home as a open resolver to have the option to anonymize or remove client IP's.

**Comment by Justin991q** [comment]
It would be nice to disable logging for a client, since I have a lot of "spam" requests from Home-Assistant which makes the query logs unreadable.

An option like this would be nice:
![image](https://user-images.githubusercontent.com/4533867/78511730-518e3100-779f-11ea-9436-bb6e737ffe7f.png)

**Comment by ArchiveBase** [comment]
> It would be nice to disable logging for a client, since I have a lot of "spam" requests from Home-Assistant which makes the query logs unreadable.

Just no too despite about disable logging for one or several clients but then you will never know what happening on that device or about any malicious traffic receive to your network. Also possible for someone to take advantage with such built in features to bypass as well. I forgot mention here about log filtering options; that will fulfill you needs https://github.com/AdguardTeam/AdGuardHome/issues/481

**Comment by Justin991q** [comment]
> > It would be nice to disable logging for a client, since I have a lot of "spam" requests from Home-Assistant which makes the query logs unreadable.
> 
> Just no too despite about disable logging for one or several clients but then you will never know what happening on that device or about any malicious traffic receive to your network. Also possible for someone to take advantage with such built in features to bypass as well. I forgot mention here about log filtering options; that will fulfill you needs #481

What about an option to hide it from the query log by default?
And if you want to see it you can just filter on that client.

But I think that has something to do with #481

**Comment by ArchiveBase** [comment]
> What about an option to hide it from the query log by default?
> And if you want to see it you can just filter on that client.
> 
> But I think that has something to do with #481

That also good idea.

**Comment by ArchiveBase** [comment]
Duplicate #767
@ameshkov suggest to close #767 instead.

**Comment by vipinjn24** [comment]
this should be there, if we are deploying it on kubernetes with liveness probe with lookup command to ensure liveness then it is required, it is flooding the logs.

**Comment by ihipop** [comment]
any update on this issue?

**Comment by dynosu** [comment]
Would like this issue to prevent health checks from coming up. Another option is to point 1 specific address as a health check which doesnt show up by default. This would be a real domain, and should resolve, but wouldnt show up in logs.

**Comment by felipecrs** [comment]
I have the exact same issue thanks to my Unifi switch:

![image](https://user-images.githubusercontent.com/29582865/220917026-97ba55cd-8232-49c5-91b5-dc4f81ef27a1.png)

**Comment by dynosu** [comment]
> I have the exact same issue thanks to my Unifi switch:
> 
> ![image](https://user-images.githubusercontent.com/29582865/220917026-97ba55cd-8232-49c5-91b5-dc4f81ef27a1.png)

Yes, i have that with the health check from my dnsdist. I need it to work or need another way to check for dnsdist if resolving still works.

**Comment by FinlayDaG33k** [comment]
The feature would still be very much appreciated.  
Landline kinda driving me nuts.

![afbeelding](https://github.com/AdguardTeam/AdGuardHome/assets/13424273/83ee81fd-6386-45c1-b4af-b7208b2aa95a)

**Comment by VsevolodNaumov** [comment]
Thanks for taking the time to report this.
I believe this is already being tracked in #767, so I'm going to close this issue as a duplicate to keep all context in one thread. If you notice anything that is not covered there, please leave a comment in #767 and we'll take it into account.
